### PR TITLE
SILGen Tests

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @aschwaighofer @jrose-apple suggestions as to why there are differences between the platforms here?

File: test/SILGen/Inputs/usr/include/CoreData.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@interface NSEntityDescription
```
Comment (jrose-apple) [inline test/SILGen/Inputs/usr/include/CoreData.h]: Probably best to import Foundation and inherit from NSObject too. Things unexpectedly being base classes does weird things in Objective-C.

**Review by jrose-apple — COMMENTED** [review]
My guess is always that you ended up using the mock SDK with the real overlays. Don't forget that the fake overlays have to be built and put into the search path!

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Comment by compnerd** [comment]
Darwin Failing Tests:

Swift(macosx-x86_64).SILGen.without_actually_escaping_block.swift
Swift(macosx-x86_64).SILGen.metatype_in_init_delegation.swift
Swift(macosx-x86_64).SILGen.objc_bridging.swift
Swift(macosx-x86_64).IRGen.newtype.swift
Swift(macosx-x86_64).IDE.newtype.swift

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**